gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -290,7 +313,16 @@ private ManagedChannel createSingleChannel() throws IOException {
       builder.keepAliveTimeout(DIRECT_PATH_KEEP_ALIVE_TIMEOUT_SECONDS, TimeUnit.SECONDS);
       builder.defaultServiceConfig(directPathServiceConfig);
     } else {
-      builder = ManagedChannelBuilder.forAddress(serviceAddress, port);
+      try {
+        ChannelCredentials mTlsChannelCredentials = createMtlsChannelCredentials();
+        if (mTlsChannelCredentials != null) {
+          builder = Grpc.newChannelBuilder(endpoint, mTlsChannelCredentials);
+        } else {
+          builder = ManagedChannelBuilder.forAddress(serviceAddress, port);
+        }
+      } catch (GeneralSecurityException e) {
```

Comment:
no need to wrap this here, and better not to. Leave the semantically correct exception in place as long as possible, ideally until it's handled. You might need to wrap it higher up the stack, but ideally don't do that unless absolutely necessary. This is not an I/O Exception.

Reply:
Done. Now wrapped it in the caller `ChannelFactory.createSingleChannel`, it is the last place to wrap it since `ChannelFactory` is an interface and `createSingleChannel` can only throw `IOException`.